core/src/client/async_client/mod.rs

```diff
@@ -186,16 +197,18 @@ impl ClientBuilder {
 		let (to_back, from_front) = mpsc::channel(self.max_concurrent_requests);
 		let (err_tx, err_rx) = oneshot::channel();
 		let max_notifs_per_subscription = self.max_notifs_per_subscription;
+		let (on_close_tx, on_close_rx) = oneshot::channel();
 
 		wasm_bindgen_futures::spawn_local(async move {
-			background_task(sender, receiver, from_front, err_tx, max_notifs_per_subscription, None).await;
+			background_task(sender, receiver, from_front, err_tx, max_notifs_per_subscription, None, on_close_tx).await;
 		});
 		Client {
 			to_back,
 			request_timeout: self.request_timeout,
 			error: Mutex::new(ErrorFromBack::Unread(err_rx)),
 			id_manager: RequestIdManager::new(self.max_concurrent_requests, self.id_kind),
 			max_log_length: self.max_log_length,
+			notify: Mutex::new(Some(on_close_rx)),
```

Comment:
no more `tokio::sync::Notify`?
doesn't it compile for WASM or what's the reason?

Reply:
This perhaps?: https://github.com/paritytech/jsonrpsee/pull/837#discussion_r938031389

Reply:
Yep, that's the explanation, mainly to reduce dependencies on tokio 🙏